supabase/migrations/20241014183452_create_investment.sql

```diff
@@ -0,0 +1,70 @@
+
+create table "public"."investment" (
+    "id" bigint generated by default as identity not null,
+    "created_at" timestamp with time zone not null default now(),
+    "last_edited_at" timestamp with time zone default now(),
+    "portfolio" bigint not null,
+    "apy" numeric,
+    "entry_fee" numeric,
+    "exit_fee" numeric,
+    "success_fee" numeric,
+    "management_fee" numeric,
+    "exit_fee_type" text,
+    "management_fee_type" text,
+    "entry_fee_type" text,
+    "name" text not null
+);
+
+
+CREATE UNIQUE INDEX investment_pkey ON public.investment USING btree (id);
+
+alter table "public"."investment" add constraint "investment_pkey" PRIMARY KEY using index "investment_pkey";
+
+alter table "public"."investment" add constraint "investment_portfolio_fkey" FOREIGN KEY (portfolio) REFERENCES portfolio(id) not valid;
+
+alter table "public"."investment" validate constraint "investment_portfolio_fkey";
+
+grant delete on table "public"."investment" to "anon";
+
+grant insert on table "public"."investment" to "anon";
+
+grant references on table "public"."investment" to "anon";
+
+grant select on table "public"."investment" to "anon";
+
+grant trigger on table "public"."investment" to "anon";
+
+grant truncate on table "public"."investment" to "anon";
+
+grant update on table "public"."investment" to "anon";
+
+grant delete on table "public"."investment" to "authenticated";
+
+grant insert on table "public"."investment" to "authenticated";
+
+grant references on table "public"."investment" to "authenticated";
+
+grant select on table "public"."investment" to "authenticated";
+
+grant trigger on table "public"."investment" to "authenticated";
+
+grant truncate on table "public"."investment" to "authenticated";
+
+grant update on table "public"."investment" to "authenticated";
+
+grant delete on table "public"."investment" to "service_role";
+
+grant insert on table "public"."investment" to "service_role";
+
+grant references on table "public"."investment" to "service_role";
+
+grant select on table "public"."investment" to "service_role";
+
+grant trigger on table "public"."investment" to "service_role";
+
+grant truncate on table "public"."investment" to "service_role";
+
+grant update on table "public"."investment" to "service_role";
+
+
+
```

Comment:
It's missing the realtime so it does not work out of the box because in adapters/supabase/index.ts we subscribe to a realtime channel.
```suggestion

alter publication supabase_realtime add table public.investment;

```